src/ser/serde.rs

```diff
@@ -551,8 +607,12 @@ impl Serialize for Decimal128 {
     where
         S: ser::Serializer,
     {
-        let value = Bson::Decimal128(self.clone());
-        value.serialize(serializer)
+        let mut state = serializer.serialize_struct("$numberDecimal", 1)?;
```

Comment:
When the `decimal128` flag was enabled, this used to produce actual extended JSON format (i.e. `{ "$numberDecimal": <decimal as a string> }`), but per the bug we discussed in triage, the current implementation's `to_string` is lossy. For that reason, it seems safer to produce this unique format that does round trip properly until we have a complete decimal128 implementation.

For users relying on the output of this, their code will break, but that's probably better than them silently losing data.
Also note that we classify the `decimal128` feature flag as "experimental" and subject to breaking changes, so this is within our contract. Code that omits the flag will not break.